lib/fastlane/plugin/wpmreleasetoolkit/actions/common/upload_to_s3.rb

```diff
@@ -0,0 +1,107 @@
+require 'fastlane/action'
+require 'digest/sha1'
+
+module Fastlane
+  module Actions
+    module SharedValues
+      S3_UPLOADED_FILE_PATH = :S3_UPLOADED_FILE_PATH
+    end
+
+    class UploadToS3Action < Action
+      def self.run(params)
+        file_path = params[:file]
+        file_name = File.basename(file_path)
+        UI.user_error!("Unable to read file at #{file_path}") unless File.file?(file_path)
+
+        bucket = params[:bucket]
+        key = params[:key]
+
+        if params[:auto_prefix] == true
+          file_name_hash = Digest::SHA1.base64digest(file_name)
+          key = [file_name_hash, key].join('/')
+        end
+
+        UI.user_error!("File already exists at #{key}") if file_is_already_uploaded?(bucket, key)
+
+        UI.message("Uploading #{file_path} to: #{key}")
+
+        File.open(file_path, 'rb') do |file|
+          Aws::S3::Client.new().put_object({
+                                             body: file,
+                                             bucket: bucket,
+                                             key: key
+                                           })
```

Comment:
Indentation nitpick.

```suggestion
          Aws::S3::Client.new().put_object(
            {
              body: file,
              bucket: bucket,
              key: key
            }
          )
```

Reply:
Or, more compact:

```suggestion
          Aws::S3::Client.new().put_object({ body: file, bucket: bucket, key: key })
```

Reply:
I even think, given how Ruby works and interprets named parameters at call site vs Hashes at def site, that you should be able to do:
```suggestion
          Aws::S3::Client.new().put_object(body: file, bucket: bucket, key: key)
```
At least that's how most of those typical Ruby client implementations (e.g. Octokit for GitHub API client) expect you to call those methods anyway 🙂

Reply:
Yep – it's funny, I reviewed another PR and suggested dropping the hash syntax, but somehow missed it here. I've simplified it to just the regular named parameter syntax (though I retained the multi-line call in case of future changes).